src/main/java/com/smartcar/sdk/SmartcarException.java

```diff
@@ -104,14 +118,73 @@ public SmartcarException(final Status status) {
    * @param status a status associated with the exception
    */
   public SmartcarException(final String message, final Status status) {
-    super(message);
+    this(message, status, null, null);
+  }
+
+  /**
+   * Initializes a new Smartcar API exception with the specified message and
+   * status.
+   *
+   * @param request the associated HTTP request
+   * @param response the associated HTTP response
+   */
+  public SmartcarException(final Request request, final Response response) {
```

Comment:
Where do these different constructors get used?

Reply:
I found including the request/response objects was useful while debugging. I think some of these are left over and no longer used though. Will prune those out.